go/store/nbs/cmp_chunk_table_writer_test.go

```diff
@@ -0,0 +1,151 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nbs
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/liquidata-inc/dolt/go/store/atomicerr"
+	"github.com/liquidata-inc/dolt/go/store/chunks"
+	"github.com/liquidata-inc/dolt/go/store/hash"
+)
+
+func TestBlockBufferTableSink(t *testing.T) {
+	suite.Run(t, &TableSinkSuite{sink: NewBlockBufferTableSink(128)})
+}
+
+func TestFixedBufferTableSink(t *testing.T) {
+	suite.Run(t, &TableSinkSuite{sink: NewFixedBufferTableSink(make([]byte, 32*1024))})
+}
+
+type TableSinkSuite struct {
+	sink ByteSink
+	t    *testing.T
+}
+
+func (suite *TableSinkSuite) SetT(t *testing.T) {
+	suite.t = t
+}
+
+func (suite *TableSinkSuite) T() *testing.T {
+	return suite.t
+}
+
+func (suite *TableSinkSuite) TestWrite() {
+	data := make([]byte, 64)
+	for i := 0; i < 64; i++ {
+		data[i] = byte(i)
+	}
+
+	for i := 0; i < 32; i++ {
+		_, err := suite.sink.Write(data)
+		assert.NoError(suite.t, err)
+	}
+}
+
+func TestCmpChunkTableWriter(t *testing.T) {
+	// Put some chunks in a table file and get the buffer back which cotains the table file data
+	ctx := context.Background()
+
+	expectedId, buff, err := WriteChunks(testMDChunks)
+	require.NoError(t, err)
+
+	// Setup a TableReader to read compressed chunks out of
+	ti, err := parseTableIndex(buff)
+	require.NoError(t, err)
+	tr := newTableReader(ti, tableReaderAtFromBytes(buff), fileBlockSize)
+
+	hashes := make(hash.HashSet)
+	for _, chnk := range testMDChunks {
+		hashes.Insert(chnk.Hash())
+	}
+
+	ae := atomicerr.New()
+	wg := &sync.WaitGroup{}
+	reqs := toGetRecords(hashes)
+	found := make(chan CompressedChunk, 128)
+
+	go func() {
+		defer close(found)
+		tr.getManyCompressed(ctx, reqs, found, wg, ae, &Stats{})
+		wg.Wait()
+	}()
+
+	// for all the chunks we find, write them using the compressed writer
+	tw := NewCmpChunkTableWriter()
+	for cmpChnk := range found {
+		err = tw.AddCmpChunk(cmpChnk)
+		require.NoError(t, err)
+	}
+
+	require.NoError(t, ae.Get())
+
+	id, err := tw.Finish()
+	require.NoError(t, err)
+
+	assert.Equal(t, expectedId, id)
+
+	output := bytes.NewBuffer(nil)
+	err = tw.Flush(output)
+	require.NoError(t, err)
+
+	outputBuff := output.Bytes()
+	outputTI, err := parseTableIndex(outputBuff)
+	require.NoError(t, err)
+	outputTR := newTableReader(outputTI, tableReaderAtFromBytes(buff), fileBlockSize)
+
+	compareContentsOfTables(t, ctx, hashes, tr, outputTR)
+}
+
+func compareContentsOfTables(t *testing.T, ctx context.Context, hashes hash.HashSet, expectedRd, actualRd tableReader) {
+	expected, err := readAllChunks(ctx, hashes, expectedRd)
+	require.NoError(t, err)
+	actual, err := readAllChunks(ctx, hashes, actualRd)
+	require.NoError(t, err)
+
+	assert.Equal(t, len(expected), len(actual))
```

Comment:
You don't need this check, the following one handles length as well as content

Reply:
You are right, I don't need it, but I think it's faster to see what went wrong, and provides disambiguation of what went wrong.